buildattr64/buildattr64.rst

```diff
@@ -0,0 +1,904 @@
+because the set of instructions conforming to architecture Armv8.0-A is
+a subset of the set conforming to architecture Armv9.0-A. Stated more
+precisely, it is the case that {ISA\@Armv8.0-A} ⊆ {ISA\@Armv9.0-A}.
+
+This partial order of the attributes often differs from the arithmetic
+order of the enumerated values of the tag. In many cases the partial
+order is:
+
+*  Identical to the arithmetic order  (as with ``Tag_THUMB_ISA_use`` in
+   (ADDENDA32_)).
+
+*  Reversed from the arithmetic order (as with ``Tag_Feature_GCS`` in
+   `aeabi-feature-and-bits subsection`_).
+
+*  Represents mutually incompatible choices with which only the identical
+   choice, or no use at all, is compatible (as with ``Tag_ABI_PCS_wchar_t`` in
+   (ADDENDA32_).
+
+Note that the appropriate partial order to use can evolve over time as
+the underlying specifications evolve.
+
+Combining two values of the same tag
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Using a1 <> a2 to denote if a1 and a2 are unordered in the partial
+order of demands/compatibility and a1 + a2 to denote the combination
+of a1 and a2. The following combination rules apply:
+
+*  If a1 ≤ a2, a1 + a2 = a2, else if a2 ≤ a1,
+   a1 + a2 = a1. (‘+’ behaves like the *maximum* function).
+
+*  If a1 <> a2 there are two mutually exclusive sub-cases.
+
+   *  There is a least a3 such that a1 ≤ a3 and a2 ≤ a3.
+      Then a1 + a2 = a3.
+
+      Example: ``Tag_CPU_arch`` from (ADDENDA32_) when a1 = v6KZ, a2 = v6T2, and a3 = v7.
+
+   *  There is no such a3, so a1 + a2 denotes the attempted combination of
+      incompatible values.
+
+      Example: ``Tag_ABI_PCS_wchar_t`` from (ADDENDA32_) when a1 = 2 and a2 = 4.
+
+In this second sub-case it is a matter of notational taste whether
+a1 + a2 is defined to have a value such as error or Top, or defined to
+have no value. Either way, in practice an attempted combination is
+expected to fail in a way specific to a tool chain’s compatibility
+model (for example by provoking a link-time diagnostic).
+
+Forcing Functions
+^^^^^^^^^^^^^^^^^
+
+In many cases the value of an attribute is independent of the value
+of all other attributes. There can be cases where attribute values
+are linked. For example to describe the signing schema in (PAuthABI64_)
+which is described as a tuple (vendor, version) two attributes are
+required. Incompatibilites also exist, such as in (ADDENDA32_)
+``Tag_Advanced_SIMD_arch`` cannot be set if ``Tag_CPU_arch_profile``
+is set to 'M'.
+
+The specification will note known forcing functions. Implementation of
+forcing functions is Q-o-I.
+
+Representing build attributes in a relocatable ELF Object file
+==============================================================
+
+Encoding
+--------
+
+Encoding build attributes are encoded in a vendor-specific section of
+type ``SHT_AARCH64_ATTRIBUTES`` and name ``.ARM.attributes`` (for
+further details see [AAELF64_]).
+
+An attribute is encoded in a <tag, value> pair.
```

Comment:
A tag with the same name in a different .aeabi_subsections is considered a different tag.